From b473ac2375f9b2c28fc44dd469a79dd9bacfa101 Mon Sep 17 00:00:00 2001
From: Markus Theil <theil.markus@gmail.com>
Date: Wed, 25 Oct 2023 13:11:52 +0200
Subject: [PATCH 2/3] fix mstprivhost.py

Signed-off-by: Markus Theil <theil.markus@gmail.com>
---
 mlxconfig/mstprivhost.py | 22 ++++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/mlxconfig/mstprivhost.py b/mlxconfig/mstprivhost.py
index 6381473..52c86fd 100644
--- a/mlxconfig/mstprivhost.py
+++ b/mlxconfig/mstprivhost.py
@@ -108,12 +108,22 @@ class PrivilegeException(Exception):
 
 
 class PrivilegeMgr(object):
-    CONFIG_CMD_LINE = "mlxconfig -d %s -f %s --yes set_raw"
-    QUERY_CMD_LINE = "mlxconfig -d %s -f %s --yes get_raw"
-    QUERY_MMHI_CMD_LINE = "mlxreg -d %s --reg_name MMHI --get"
-    QUERY_NVGC_CMD_LINE = "mlxreg -d %s --reg_name MNVGC --get"
-    QUERY_HW_ACCESS_STATUS = "mlxreg -d %s --reg_id 0x402d --reg_len 0x4 --get"
-    MCRA_CMD_LINE = "mcra %s 0xf0014.0:16"
+    IS_MSTFLINT = os.path.basename(__file__) == "mstprivhost.py"
+
+    MCRA = 'mcra'
+    MLXCONFIG = 'mlxconfig'
+    MLXREG = 'mlxreg'
+    if IS_MSTFLINT:
+        MLXCONFIG = 'mstconfig'
+        MLXREG = 'mstreg'
+        MCRA = "mstmcra"
+
+    CONFIG_CMD_LINE = f"{MLXCONFIG} -d %s -f %s --yes set_raw"
+    QUERY_CMD_LINE = f"{MLXCONFIG} -d %s -f %s --yes get_raw"
+    QUERY_MMHI_CMD_LINE = f"{MLXREG} -d %s --reg_name MMHI --get"
+    QUERY_NVGC_CMD_LINE = f"{MLXREG} -d %s --reg_name MNVGC --get"
+    QUERY_HW_ACCESS_STATUS = f"{MLXREG} -d %s --reg_id 0x402d --reg_len 0x4 --get"
+    MCRA_CMD_LINE = f"{MCRA} %s 0xf0014.0:16"
     TLV_DATA_RE = re.compile(r'Data\s*:\s*(.+)\n', flags=re.MULTILINE)
 
     BLUE_FIELD_DEV_ID = 0x211
-- 
2.42.0

